coops-ui/src/components/Dashboard/FilingHistoryList.vue

```diff
@@ -196,17 +210,57 @@ export default {
           filingAuthor: filing.changeOfAddress.certifiedBy,
           filingDate: filing.header.date,
           filingStatus: filing.header.status,
-          filingDocuments: [{ name: 'Address Change', url: '' }]
+          filingDocuments: [{
+            filingId: filing.header.filingId,
+            name: 'Address Change',
+            documentName: 'AddressChange - ' + filing.header.date
+          }]
         }
         this.filedItems.push(item)
       } else {
         console.log('ERROR - invalid changeOfAddress in filing =', filing)
       }
     },
 
+    loadDocument (filingDocument) {
+      var url = this.corpNum + '/filings/' + filingDocument.filingId
+      let headers = { 'Accept': 'application/pdf' }
+
+      axios.get(url, { headers: headers, responseType: 'arraybuffer' }).then(response => {
+        if (response) {
+          /* solution from https://github.com/axios/axios/issues/1392 */
+
+          // It is necessary to create a new blob object with mime-type explicitly set
+          // otherwise only Chrome works like it should
+          var newBlob = new Blob([response.data], { type: 'application/pdf' })
+
+          // IE doesn't allow using a blob object directly as link href
+          // instead it is necessary to use msSaveOrOpenBlob
+          if (window.navigator && window.navigator.msSaveOrOpenBlob) {
+            window.navigator.msSaveOrOpenBlob(newBlob)
+            return
+          }
+
+          // For other browsers:
+          // Create a link pointing to the ObjectURL containing the blob.
+          const data = window.URL.createObjectURL(newBlob)
+          var link = document.createElement('a')
+          link.href = data
+          link.download = filingDocument.documentName
+          link.click()
+        } else {
+          this.errorMessage = 'Oops, could not load data from server'
+        }
+      }).catch(error => {
+        console.error('loadDocument() error =', error)
+        this.errorMessage = 'Oops, could not load data from server'
+      })
+    },
+
     downloadAll (item) {
-      // TODO
-      console.log('downloadAll(), item =', item)
+      for (let i = 0; i < item.filingDocuments.length; i++) {
```

Comment:
I think TypeScript does allow `for (let document of item.filingDocuments)` (FWIW).